Refactor metric spaces (#1450)
UniMath/SymmetryBookFormalization: src/metric-spaces/cauchy-approximations-metric-spaces.lagda.md

A
{{#concept "Cauchy approximation" Disambiguation="in a metric space" Agda=is-cauchy-approximation-Metric-Space}}
in a [metric space](metric-spaces.metric-spaces.md) `A` is a map `f` from
[`ℚ⁺`](elementary-number-theory.positive-rational-numbers.md) to the carrier

> type of `A` such that for all `(ε δ : ℚ⁺)`, `f ε` and `f δ` are in a

Reviewer: ```suggestion
type of `A` such that for all [positive rationals](elementary-number-theory.positive-rational-numbers.md) `ε` and `δ`, `f ε` and `f δ` are in a
```

A
{{#concept "Cauchy approximation" Disambiguation="in a metric space" Agda=is-cauchy-approximation-Metric-Space}}
in a [metric space](metric-spaces.metric-spaces.md) `A` is a
[Cauchy approximation](metric-spaces.cauchy-approximations-pseudometric-spaces.md)

> in its underlying [pseudometric-space](metric-spaces.pseudometric-spaces.md): a

Reviewer: ```suggestion
in its underlying [pseudometric space](metric-spaces.pseudometric-spaces.md): a
```

  is-cauchy-approximation-prop-Metric-Space :
    (ℚ⁺ → type-Metric-Space A) → Prop l2
  is-cauchy-approximation-prop-Metric-Space f =
    Π-Prop
      ( ℚ⁺)
      ( λ ε →
        Π-Prop
          ( ℚ⁺)
          ( λ δ →

>             neighborhood-prop-Metric-Space A (ε +ℚ⁺ δ) (f ε) (f δ)))

Reviewer: ```suggestion
  is-cauchy-approximation-prop-Metric-Space f =
    is-cauchy-approximation-prop-Pseudometric-Space
      ( pseudometric-space-Metric-Space A)
```

>   is-cauchy-approximation-prop-Metric-Space f =
>     Π-Prop
>       ( ℚ⁺)
>       ( λ ε →
>         Π-Prop
>           ( ℚ⁺)
>           ( λ δ →
>             neighborhood-prop-Metric-Space A (ε +ℚ⁺ δ) (f ε) (f δ)))

Reviewer: ```suggestion
  is-cauchy-approximation-prop-Metric-Space =
    is-cauchy-approximation-prop-Pseudometric-Space
      ( pseudometric-space-Metric-Space A)
```

  const-cauchy-approximation-Metric-Space :
    cauchy-approximation-Metric-Space A
  const-cauchy-approximation-Metric-Space =
    (const ℚ⁺ x) , (λ ε δ → refl-neighborhood-Metric-Space A (ε +ℚ⁺ δ) x)
```

> ### Short maps between metric spaces are functorial on Cauchy approximations

Reviewer: ```suggestion
### The action of short maps on Cauchy approximations
```
